doc/MAINTAINER_GUIDE.md

```diff
@@ -437,3 +437,33 @@ GHC 7.8.4 is slightly different:
     sed -i 's/^TAR_COMP *= *bzip2$/TAR_COMP = xz/' mk/config.mk && \
     make -j$(cat /proc/cpuinfo|grep processor|wc -l) && \
     make binary-dist
+
+## Upgrading msys
+
+When installing GHC on Windows, Stack will also install
+[msys2](http://www.msys2.org/) to provide a Unix shell and environment,
+necessary for such things as running configure scripts. This document explains
```

Comment:
I think this PR is basically good to go and will probably save folks a lot of time later on, but it may be good to tweak:

> This document explains the steps to take to upgrade the msys2 version used by Stack.

to instead read as:

> This section explains the steps required to upgrade the msys2 version used by Stack.

I know the "document" part is there because this blurb was originally meant to be in a separate file.

Reply:
That's much nicer, thanks! PR updated.